bindings/python/tests/test_parse.py

```diff
@@ -32,3 +32,75 @@ def get_parsed_info(tmp_path, file_contents):
     )
 
     return parsed_info
+
+
+def test_return_type_composition(tmp_path):
+    file_contents = ""
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    assert type(parsed_info) is dict
+    assert type(parsed_info["members"]) is list
+    assert len(parsed_info["members"]) == 0
+
+
+def test_translation_unit(tmp_path):
+    file_contents = ""
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    assert parsed_info["kind"] == "TRANSLATION_UNIT"
+    assert parsed_info["depth"] == 0
+    assert parsed_info["line"] == 0
+    assert parsed_info["column"] == 0
+    assert parsed_info["name"] == str(tmp_path / "file.hpp")
+
+
+def test_namespace(tmp_path):
+    file_contents = "namespace a_namespace {}"
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    namespace = parsed_info["members"][0]
+
+    assert namespace["kind"] == "NAMESPACE"
+    assert namespace["name"] == "a_namespace"
+
+
+def test_namespace_ref(tmp_path):
+    file_contents = """
+    #include <ostream>
+    std::ostream anOstream;
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    var_decl = parsed_info["members"][0]
+    namespace_ref = var_decl["members"][0]
+
+    assert namespace_ref["kind"] == "NAMESPACE_REF"
+    assert namespace_ref["name"] == "std"
+
+
+def test_var_decl(tmp_path):
+    file_contents = "int anInt = 1;"
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    var_decl = parsed_info["members"][0]
+
+    assert var_decl["kind"] == "VAR_DECL"
+    assert var_decl["element_type"] == "Int"
+    assert var_decl["name"] == "anInt"
+
+
+def test_call_expr(tmp_path):
+    file_contents = """
+    int aFunction() {
+        return 1;
+    }
+    int anInt = aFunction();
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    var_decl = parsed_info["members"][1]
+    call_expr = var_decl["members"][0]
+
+    assert call_expr["kind"] == "CALL_EXPR"
+    assert call_expr["name"] == "aFunction"
```

Comment:
Could you verify that the call expr goes into variable `anInt`?

Reply:
It is a member of `VAR_DECL`, so there's that:
```diff
- ...
- ...
{
    "depth": 1,
    "line": 5,
    "column": 9,
    "kind": "VAR_DECL",
    "name": "anInt",
    "element_type": "Int",
    "members": [
        {
            "depth": 2,
            "line": 5,
            "column": 17,
            "kind": "CALL_EXPR",
            "name": "aFunction",
            "element_type": "Int",
            "members": [
                {
                    "depth": 3,
                    "line": 5,
                    "column": 17,
                    "kind": "UNEXPOSED_EXPR",
                    "name": "aFunction",
                    "element_type": "Pointer",
                    "members": [
                        {
                            "depth": 4,
                            "line": 5,
                            "column": 17,
                            "kind": "DECL_REF_EXPR",
                            "name": "aFunction",
                            "element_type": "FunctionProto",
                            "members": [],
                        }
                    ],
                }
            ],
        }
    ],
}
- ...
- ...
```

If you meant that `anInt` gets initialised, I don't think we can do that, since libclang is a semantic parser.

Reply:
I meant add `assert var_decl["name"] = "anInt"`

Reply:
Ah. Done.